src/pages/Artwork.js

```diff
@@ -8,12 +8,28 @@ const Artwork = () => {
   const [artwork, setArtwork] = useState(null);
   const { id } = useParams();
 
+  const historialStorage = useCallback( 
+    () => {
+      let array = [...JSON.parse(localStorage.getItem("historial") || "[]")];
+      const { title, images } = artwork;
+
+      if (array.some(elem => elem.id === id )) return null;
+
+      array.length > 4 && array.shift();
```

Comment:
I don't get the `shift` use here, please explain

Reply:
It's like not adding more than 4 items to the list?

Reply:
![limit](https://user-images.githubusercontent.com/71513679/162526689-49ee4b43-f396-42e6-91ec-f227d39ba123.gif)

This line removes the first element if there are more than 4, then adds one more. Always limited to 5